src/modules/mc_pos_control/Takeoff/Takeoff.cpp

```diff
@@ -74,6 +75,9 @@ void TakeoffHandling::updateTakeoffState(const bool armed, const bool landed, co
 			_takeoff_state = TakeoffState::rampup;
 			_takeoff_ramp_progress = 0.f;
 
+		} else if (!landed) {
```

Comment:
@MaEtUgR Is that fine?

Reply:
@MaEtUgR , to give you some context here: before the change, the position controller to start working required the "want_takeoff" to be true, which means that a pilot needed to put the throttle up. It resulted in a behavior that I found counterintuitive when we put the drone directly into the air after arming: in other modes we tested (hold, stabilized, offboard) the drone behaved as expected (i.e., just started flying), but in position, if we forgot to raise the throttle up, the drone would just fall down, slightly spinning the propellers.

This change does not require the "want_takeoff" to happen if the drone is thrown in the air after arming.

Reply:
But that means if you run any drone and kick it on the ground it will start to correct the position in place and tip over.

Reply:
The case where you arm in other modes and later on switch to a climb rate enabled mode is handled with this: https://github.com/PX4/PX4-Autopilot/blob/23cf0a7317805528d8cdb57806efff83b87895f0/src/modules/mc_pos_control/MulticopterPositionControl.cpp#L564-L566

Can't the case where you throw the drone not be handled explicitly in that way? E.g. when you're in throw mode you are idling ready in the "about to be thrown" state and jump to enabling all controllers skipping the takeoff ramp in an explicit place.

Compared to the suggested whenever a multicopter is idling on the ground and moves a bit too much it jumps up.

Reply:
My main motivation to choose what is proposed was that for me the fact that the drone does not start regulating position after finding itself in the air was counterintuitive (we actually crashed twice before we realized what's happening). We've never had problems with drones starting flying unexpectedly with this firmware, but we treat all armed drones with care, so it never happened that a drone was moved/kicked after being armed.

Anyway, I do understand that we might not want to change the existing behavior to implement the throw launch feature. However, as throw launch is not implemented as a separate mode, the line you pointed out does not automatically trigger, so some modification is required. Do you maybe have some suggestions on how to do that without introducing any mode switching?

The option I was considering was to modify the MulticopterPositionControl and add an if statement checking if THROW_LANUCH parameter is enabled. If so, explicitly skip the takeoff. Is that something you find acceptable?